src/components/BreedDetails.vue

```diff
@@ -42,12 +41,33 @@ export default {
       type: Object,
       required: true,
     },
+
+    searchQuery: {
+      type: String,
+      required: false,
+    },
   },
+
   computed: {
     hiddenClass() {
       return this.breed.isActive ? "result__details-page" : "hidden";
     },
   },
+  methods: {
+    goBackToPreviousRoute() {
+      if (this.$route.params.prevRoute === "/breeds/search") {
+        this.$router.push({
+          name: "search",
+          params: {
+            search: this.searchQuery,
+          },
+          query: { q: this.searchQuery },
+        });
+      } else {
+        this.$router.push({ name: "breeds" });
+      }
+    },
+  },
```

Comment:
- Dodal sem metodo `goBackToPreviousRoute`. Če kliknem Back btn od `DetailsPage `te odpelje nazaj na prejšnji route in sicer sem v `BreedCard `komponenti dodal previosRoute param na router link

`<router-link
        :to="{
          name: 'details',
          params: { breedName: breed.name, prevRoute: $route.path },
        }"
        >Learn more...</router-link
      >`
     

- Dodal sem keep-alive komponento, ker če zamenjam  route ko vpisujem v input, se ta vrednost iz inputa ne shrani, ampak se izbriše

Reply:
pomoje ima tud router opcije da greš za eno routo nazaj.. poglej mal tukej -> https://router.vuejs.org/guide/essentials/navigation.html

je pa možno da je tud povezan z router `mode: 'history'` nastavitvijo